### DISCUSSION TITLE
Message of 'AutoMapperMappingException' should contain concrete/runtime type name

### DISCUSSION THREAD

**Discussion by Der-Kraken** [LuckyPennySoftware/AutoMapper#3657] [NA]
If I map but missing a configuration in profile then the **AutoMapperMappingException** is thrown. The output contains the name of source and target type which is a good thing. But the type name is only the build-time type not the runtime one. See my example below:

```csharp
public interface ITableModel { }
public interface IDtoModel { }
public class ConcreteTableModel : ITableModel { }
public class ConcreteDtoModel : IDtoModel { }

public class MyProfile : Profile { }

class Program
{
    static void Main(string[] args)
    {
        var configuration = new MapperConfiguration(cfg => new MyProfile());
        var mapper = new AutoMapper.Mapper(configuration);

        // The error message shows the declared type (left side).
        // I propose to show the runtime type (right side).
        ITableModel concreteTableModel = new ConcreteTableModel();
        IDtoModel concreteDtoModel = new ConcreteDtoModel();

        try
        {
            mapper.Map(concreteTableModel, concreteDtoModel);
        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
            Console.ReadLine();
        }
    }
}
```
## Current Message of 'AutoMapperMappingException'
```
Missing type map configuration or unsupported mapping.

Mapping types:
ITableModel -> IDtoModel
ConsoleApp6.ITableModel -> ConsoleApp6.IDtoModel
```
## Proposed Message of 'AutoMapperMappingException'
```
Missing type map configuration or unsupported mapping.

Mapping types:
ConcreteTableModel-> ConcreteDtoModel
ConsoleApp6.ConcreteTableModel-> ConsoleApp6.ConcreteDtoModel
```

Why I propose that? Because If I miss a mapping then I miss a concrete mapping. We do not map abstractions. If I'm missing something then maybee the error message can contain both.
Atm. I must reproduce the runtime scenario and debug it to identify the concrete type. If the message would contain it then I must only add a new mapping configuration.

**Comment by lbargaoanu** [comment]
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by Der-Kraken** [reply]
In build '10.1.2-alpha.0.4' my requirement is contained. So the proposal is done. Sorry
